clusterloader2/pkg/execservice/exec_service.go

```diff
@@ -145,6 +145,18 @@ func RunCommand(pod *corev1.Pod, cmd string) (string, error) {
 	return stdout.String(), nil
 }
 
+func RunCommandWithTimeout(pod *corev1.Pod, cmd string, timeout time.Duration) (string, error) {
+	var stdout, stderr bytes.Buffer
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
IMHO more idiomatic way to do that is "RunCommandWithContext" approach where you take context as a parameter and e.g. caller passes context with timeout or cancel function or overall something more precise that context.Background

Reply:
Hmm, I think we could do even better and just create one function RunCommand() where ctx is the first argument. I

Reply:
+1